### DISCUSSION TITLE
Would it be possible to trigger realtime messages on database changes?

### DISCUSSION THREAD

**Discussion by endlessuniverse** [centrifugal/centrifugo#655] [Ideas]
For example on PostgresDB, using something like this: https://github.com/ably-labs/ably-postgres-connector or https://github.com/ihippik/wal-listener

Edit: Sorry, first I need to ask if Centrifugo community edition is suitable for a database-driven realtime applications where long-term storage and update triggers from previously published messages are possible. More info: https://ably.com/blog/ably-postgres-connector

Thank you

**Comment by FZambia** [comment]
@endlessuniverse hello!

We currently do not have things similar to those you pointed to for PostgreSQL. One project I have seen though is https://github.com/cailloumajor/centrifugo-change-stream (for MongoDB).

I like the pattern and I suppose some Centrifugo users already do sth like this to reliably send events to Centrifugo upon changes in database. Only a guess though - no real use case information.

More integration with PostgreSQL is definitely a welcome thing for Centrifugo.

**Comment by FZambia** [comment]
Now experimenting with PostgreSQL transactional outbox, Debezium, transferring and consuming from Kafka to Centrifugo. Probably these experiments result into sth built-in.

**Reply by FZambia** [reply]
https://centrifugal.dev/docs/server/consumers + https://centrifugal.dev/docs/tutorial/outbox_cdc
